### PR TITLE
Fix Hamilton's Great Adventure

### PR DESCRIPTION
**Pull request by flash-fire**
This game calls sys_rsx_context_attribute before sys_rsx_memory_allocate.  This inevitably causes a crash.  So, I return that the resource doesn't exist.  Doesn't seem like a race condition because the main thread asks for the context before the rendering thread is even created. Could be wrong though so I left that in as a comment.

### CODE REVIEW

**Comment by digitaldude555** [comment]
screenshot 1338
yes

File: rpcs3/Emu/Cell/lv2/sys_rsx.cpp [hunk 1]
```diff
@@ -230,6 +230,13 @@ s32 sys_rsx_context_attribute(s32 context_id, u32 package_id, u64 a3, u64 a4, u6
 
 	auto m_sysrsx = fxm::get<SysRsxConfig>();
 
+	if (!m_sysrsx)
+	{
+		// Hamilton's Great Adventure [NPUB30458] calls this function before calling allocate. Not sure if race condition.
+		sys_rsx.error("sys_rsx_context_attribute called before sys_rsx_context_allocate: context_id=0x%x, package_id=0x%x, a3=0x%llx, a4=0x%llx, a5=0x%llx, a6=0x%llx)", context_id, package_id, a3, a4, a5, a6);
+		return CELL_ESRCH;
```
Comment (kd-11) [inline rpcs3/Emu/Cell/lv2/sys_rsx.cpp]: Just change to EINVAL to match real hw

Comment (flash-fire) [reply rpcs3/Emu/Cell/lv2/sys_rsx.cpp]: Fixed and rebased. Sorry I don't pay attention to github notifications.

Comment (AniLeo) [reply rpcs3/Emu/Cell/lv2/sys_rsx.cpp]: If handling that matches real console behavior then it shouldn't have the "Not sure if it's race condition" comment as it's correct

**Comment by AniLeo** [comment]
For future reference, the behavior implemented here has been tested on a real PS3 and it was confirmed correct. Merging as soon as it finishes building.

**Pull request merged**